modules/gdscript/language_server/gdscript_language_protocol.cpp

```diff
@@ -184,7 +184,9 @@ Dictionary GDScriptLanguageProtocol::initialize(const Dictionary &p_params) {
 	if (root_uri.length() && is_same_workspace) {
 		workspace->root_uri = root_uri;
 	} else {
-		workspace->root_uri = "file://" + workspace->root;
+		String r_root = workspace->root;
+		r_root = r_root.replacen(":", "%3A");
+		workspace->root_uri = "file:///" + r_root;
```

Comment:
Isn't that going to be one `/` too much on Unix?

Reply:
it shouldn't be the root_uri should be `file:///c%3A` as that is what the LSP server communicates with this then gets converted to `res://` in `get_file_path()` also made sure that `get_file_uri()` would correctly convert it back to `file:///c%3A`.

Reply:
Well `c%3A` is just `c:` which is the default root partition on Windows. I don't think that applies to Linux and macOS?

Reply:
`workspace->root` already starts with `/` on Linux and macOS, so you'll end up with `file:////` which won't work.

Reply:
ok so i see what is going on here windows will return `c%3A\\path\\to\\root` and linux will return `\path\to\root` ?